synapse/rest/client/v2_alpha/register.py

```diff
@@ -134,9 +134,11 @@ def on_POST(self, request):
             # fallback to 'username' if they gave one.
             if isinstance(body.get("user"), basestring):
                 desired_username = body["user"]
```

Comment:
would be sorely tempted to change this line to: 

```python
desired_username = body.get("user", desired_username)
```

As all the `isinstance(..., basestring)` is making me sad